src/Stratis.Bitcoin.Tests/P2P/PeerSelectorTests.cs

```diff
@@ -836,7 +836,36 @@ public void PeerSelector_ReturnConnectedPeers_AfterHandshakeFailure_HandshakeSuc
             Assert.Equal(0, peer.HandshakedAttempts);
             Assert.Null(peer.LastHandshakeAttempt);
         }
-        
+
+        [Fact]
+        public void PeerSelector_ReturnConnectedPeers_AfterThresholdExpired_ResetCounters()
+        {
+            IPAddress ipAddress = IPAddress.Parse("::ffff:192.168.0.1");
+            DataFolder peerFolder = CreateDataFolder(this);
+            PeerAddressManager peerAddressManager = this.CreatePeerAddressManager(peerFolder);
+            peerAddressManager.AddPeer(new IPEndPoint(ipAddress, 80), IPAddress.Loopback);
+
+            peerAddressManager.PeerConnected(new IPEndPoint(ipAddress, 80), DateTime.UtcNow.AddSeconds(-80));
+            DateTime firstHandshakeAttemptTime = DateTime.UtcNow.AddSeconds(-80);
+
+            PeerAddress peer = peerAddressManager.Peers[0];
+
+            // Peer selected after one handshake failure.
+            peer.SetHandshakeAttempted(firstHandshakeAttemptTime);
+            Assert.Equal(1, peer.HandshakedAttempts);
+            Assert.Contains(peer, peerAddressManager.PeerSelector.FilterBadHandshakedPeers(peerAddressManager.Peers));
+
+            // Peer selected after two handshake failures.
+            peer.SetHandshakeAttempted(firstHandshakeAttemptTime);
```

Comment:
2nd handshake but same time as first handshake - intentional? reads a bit strangely

Reply:
@thefazzer I've fixed these for you.